### PR TITLE
update libs to introduce explainer type

### PR DESCRIPTION
**Pull request by marjisound**
## What does this change?
This PR does the following:
- Updates `guardian/libs` version to `7.1.0` in all **DCR**, **AR** & **common rendering**
- fixes the compiler errors in **DCR** due to the breaking change in https://github.com/guardian/libs/releases/tag/v7.0.0 ( Thanks @mxdvl for helping out on resolving these)
- Adds `Explainer` design type in **AR** item.ts
- Ran `make gen-stories` for **DCR** as it was necessary after the new format design `Explainer`

## Why?
The new type Explainer is required as part of the new changes the Editorial Design teams are introducing. More info can be found here: 
https://docs.google.com/document/d/1MqSldxfgR86kOJaXTFNaBs5LzPzSa7jC4D4YwqqMFwg/edit#

### CODE REVIEW

File: dotcom-rendering/src/web/lib/countryCodes.ts
Comment (mxdvl) [inline dotcom-rendering/src/web/lib/countryCodes.ts:1]: This could make its way to libs!

**Review by mxdvl — APPROVED** [review]
Happy with the custom type guards, on which I paired.

**Review by JamieB-gu — APPROVED** [review]
Looks good from the AR side; I'll let Dotcom review the DCR side.

From what I can tell, at the moment if a piece is published with this tag AR will parse it correctly as an explainer. There won't, however, be a layout for it so it'll fall back to the "not implemented" page in the layout component:

https://github.com/guardian/dotcom-rendering/blob/5abc09e96401a0b9e3fcabc686aae40a6886e552/apps-rendering/src/components/Layout/index.tsx#L127

Just checking this is expected behaviour, and that we'll be implementing a layout for this before content of this type gets published? When that happens we'll also want to add a story to `Layout.stories.tsx` to get some snapshot testing.

**Comment by marjisound** [comment]
> From what I can tell, at the moment if a piece is published with this tag AR will parse it correctly as an explainer. There won't, however, be a layout for it so it'll fall back to the "not implemented" page in the layout component:


Thanks @JamieB-gu for the reminder. So I think there's a risk right now that some of the production articles with tag `tone/explainer` may be rendered by AR. So I think before merging this PR, I should either filter out explainer articles from https://github.com/guardian/mobile-apps-api/blob/main/common-play/src/main/scala/data/RenderedItemUtils.scala

or another option is, remove the part that's adding this design in item.ts in this PR and only add it back when we have completed the layout design?

**Comment by abeddow91** [comment]
How are explainer tags currently rendered? Would another option be to add Design.Explainer to the existing layout in switch in AR whilst the new template is being built?

**Comment by marjisound** [comment]
> How are explainer tags currently rendered? Would another option be to add Design.Explainer to the existing layout in switch in AR whilst the new template is being built?

I think currently, they are rendered depending on their other tags. Maybe mostly they would go through the standard layout. I think it by removing the condition `else if (isExplainer(tags))...` from item.ts, we are letting them get rendered by standard layout

**Comment by marjisound** [comment]
@JamieB-gu I added the Explainer design as part of the default Standard layout. Is that OK?

**Comment by marjisound** [comment]
[Create Explainer design type in apps rendering](https://trello.com/c/UmBENXG2/534-create-explainer-design-type-in-apps-rendering)

**Pull request merged**
